File: Analisis_data_shopee/shopee/1.r
library(RMySQL)
library(dplyr)
library(lubridate)

# Ganti 'dbname', 'username', 'password', dan 'host' sesuai dengan informasi database Anda
con <- dbConnect(MySQL(), dbname = "pembelian_shopee", user = "root", password = "", host = "localhost")

query <- "SELECT * FROM order_detail"
df <- dbGetQuery(con, query)
dbDisconnect(con)

print(head(df))

# df <- df[df$is_valid == 1, ]
# df <- filter(df, is_valid == 1)
df <- df %>% filter(is_valid == 1)

df$order_date <- as.Date(df$order_date)
# df <- mutate(df, order_date = as.Date(order_date))

# df <- df[year(df$order_date) == 2021, ]
# df <- filter(df, year(order_date) == 2021)
df <- df %>% filter(year(order_date) == 2021) 

# df <- mutate(df, month = month(order_date))
df$month <- month(df$order_date)

print(head(df))


hasil <- df %>% group_by(month) %>% summarise(total = sum(after_discount)) %>% arrange(desc(total))
print(hasil)

File: Analisis_data_shopee/shopee/1.py
import pandas as pd
import mysql.connector
import numpy as np
import matplotlib.pyplot as plt

connection = mysql.connector.connect(host='localhost', user='root', password='', database='pembelian_shopee')
query = "SELECT*FROM order_detail"
df = pd.read_sql(query, connection)
connection.close()

print(df)
df = df[df['is_valid'] == 1]
df['order_date'] = pd.to_datetime(df['order_date'])
# df = df[df['order_date'].dt.year == 2021]
# df['month'] = df['order_date'].dt.month
df = df.query('order_date.dt.year == 2021')
df = df.assign(month = df['order_date'].dt.month)
hasil = df.groupby('month')['after_discount'].sum().round().reset_index(name='jumlah').sort_values('jumlah', ascending=False).reset_index(drop=True)
print(hasil)